trin-core/src/portalnet/types.rs

```diff
@@ -345,6 +345,137 @@ impl FromStr for HexData {
     }
 }
 
+pub enum ContentKey {
+    StateContentKey(StateContentKey),
+    HistoryContentKey(HistoryContentKey),
+}
+
+pub enum StateContentKey {
+  AccountTrieNode(AccountTrieNode),
+  ContractStorageTrieNode(ContractStorageTrieNode),
+  AccountTrieProof(AccountTrieProof),
+  ContractStorageTrieProof(ContractStorageTrieProof),
+  ContractBytecode(ContractBytecode),
+}
+
+impl StateContentKey {
+  // decodes from bytes to ssz container
+  pub fn from_bytes(bytes: &[u8]) -> Result<Self, String> {
+      if let Some(content_type) = bytes.first() {
+          match content_type {
+              0x00 => Ok(StateContentKey::AccountTrieNode(
+                  AccountTrieNode::from_ssz_bytes(&bytes[1..])
+                    .map_err(|e| format!("Failed to decode ssz: {:?}", e))?
```

Comment:
Have you looked at the errors resulting from this? It seems like it might be useful to explicitly mention that the code was attempting to decode with `AccountTrieNode`, maybe maybe that's already clear enough in `e`? Also, if `e` doesn't include the byte source, I'd like to see it added here, hex-encoded. It will be so much easier to track down bugs later from the logs, if we have the source bytes and know which thing we were trying to decode as.